### PR TITLE
only mark chat as read if you're actually active

### PR DESCRIPTION
**Pull request by chrisnojima**
On osx when you close a window sibling windows will be foregrounded. There are also times where foregrounding a window can focus/defocus the main window.
This has the bad side effect in chat where this can cause you to read the currently selected thread

Steps to repro:
1. Open chat tab w/ a convo selected
2. Bring another app into focus (not devtools, else that can get focused instead)
3. Click on widget 2x

This should bring the main back on top of the app from step 2 and mark the thread read.

This PR changes the flow so you need the userActive flag (which is driven by InputMonitor) in addition to appFocus

I moved InputMonitor into the render thread so its easier to analyze and it does all window / actions stuff anyways so it doesn't really belong on the node side

- [x] Move input monitor to render thread (platform specific saga)
- [x] Remove unused appState
- [x] Never allow both timers to overlap unless we intend to

### CODE REVIEW

File: shared/actions/platform-specific/input-monitor.desktop.js [hunk 1]
```diff
@@ -1,62 +1,73 @@
 // @flow
 
+const timeToConsiderActiveForAwhile = 300000
```
Comment (chrisnojima) [inline shared/actions/platform-specific/input-monitor.desktop.js]: you can reduce these values to test how this works

File: shared/desktop/app/app-state.desktop.js [hunk 2]
```diff
@@ -19,7 +19,6 @@ export type State = {
   dockHidden: boolean,
   notifySound: boolean,
   openAtLogin: boolean,
-  isUserActive: ?boolean,
```
Comment (chrisnojima) [inline shared/desktop/app/app-state.desktop.js]: not used at all

File: shared/desktop/renderer/main.desktop.js [hunk 3]
```diff
@@ -91,28 +89,6 @@ function setupApp(store, runSagas) {
   })
   SafeElectron.getIpcRenderer().send('install-check')
 
-  var inputMonitor = new InputMonitor(function(isActive) {
```
Comment (chrisnojima) [inline shared/desktop/renderer/main.desktop.js]: moved to actions

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready
cc: @malgorithms

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/actions/platform-specific/input-monitor.desktop.js [hunk 4]
```diff
@@ -1,62 +1,73 @@
...
+    window.addEventListener('blur', this.goInactive)
+    this.goListening()
   }
 
-  startActiveTimer = () => {
-    // wait 5 minutes before adding listeners
-    if (this.activeTimeoutID) window.clearTimeout(this.activeTimeoutID)
-    this.activeTimeoutID = window.setTimeout(this.goListening, 300000)
+  _clearTimers = () => {
+    if (this.activeTimeoutID) {
```
Comment (buoyad) [inline shared/actions/platform-specific/input-monitor.desktop.js]: :sheep: I don't think these `if`s are necessary, `clearTimeout` is a noop with `null`

File: shared/actions/platform-specific/index.desktop.js [hunk 5]
```diff
@@ -78,6 +80,45 @@ const writeElectronSettingsNotifySound = (_, action) =>
...
+  const channel = Saga.eventChannel(emitter => {
+    window.addEventListener('focus', () => emitter('focus'))
+    window.addEventListener('blur', () => emitter('blur'))
+    return () => {}
+  }, Saga.buffers.expanding(1))
+  while (true) {
+    const type = yield Saga.take(channel)
+    switch (type) {
+      case 'focus':
+        if (skipAppFocusActions) {
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/actions/platform-specific/index.desktop.js]: this is the old behavior, this is so you can see it as you click in and out. if you saw it once before the flood of logs you'd likely not see it

File: shared/actions/platform-specific/index.desktop.js [hunk 6]
```diff
@@ -78,6 +80,45 @@ const writeElectronSettingsNotifySound = (_, action) =>
...
+          console.log('Skipping app focus actions!')
+        } else {
+          yield Saga.put(ConfigGen.createChangedFocus({appFocused: false}))
+        }
+    }
+  }
+}
+
+function* initializeInputMonitor(): Generator<any, void, any> {
+  const inputMonitor = new InputMonitor()
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/actions/platform-specific/index.desktop.js]: i had that originally but its useful to be able to do stuff with it outside the channel. i changed stuff back and forth but i think its fine not being closed over

File: shared/actions/platform-specific/input-monitor.desktop.js (hunk 4 above)
Comment (chrisnojima) [reply shared/actions/platform-specific/input-monitor.desktop.js]: i think flow wanted this, lemme see

File: shared/actions/platform-specific/index.desktop.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/actions/platform-specific/index.desktop.js]: Moving it before the switch would still do it on click in+out, right?

**Comment by chrisnojima** [comment]
@buoyad updated

**Pull request merged**